pkg/cmd/translator_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright © 2022 Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/translator"
+)
+
+func TestTranslate(t *testing.T) {
+	testCases := []struct {
+		name         string
+		output       string
+		resourceType translator.GWAPIResourceType
+		errorCount   int
+	}{
+		{
+			name:       "from-ingress-to-simple-gwapi",
```

Comment:
Yes, I will remove this.

Reply:
Wouldn't the expected output filenames all have the same prefix as the input filenames?

Reply:
I agree that if we're going to use `testdata/in` and `testdata/out`, we should just read files automatically. If we need multiple files, then we can move each test case to its own directory, and have the structure be `testdata/<testcase>/in` and `testdata/<testcase>/out`. That seems like it would allow us to also test handling multiple different types of file too.